routers/web/repo/tree.go

```diff
@@ -52,3 +54,29 @@ func isExcludedEntry(entry *git.TreeEntry) bool {
 
 	return false
 }
+
+func Tree(ctx *context.Context) {
+	treePath := ctx.PathParam("*")
+	ref := ctx.FormTrim("ref")
+	recursive := ctx.FormBool("recursive")
+
+	gitRepo, closer, err := gitrepo.RepositoryFromContextOrOpen(ctx, ctx.Repo.Repository)
+	if err != nil {
+		ctx.ServerError("RepositoryFromContextOrOpen", err)
+		return
+	}
+	defer closer.Close()
+
+	refName := gitRepo.UnstableGuessRefByShortName(ref)
```

Comment:
Do not guess anymore, frontend should pass the correctly ref to backend.

Reply:
Removed.